src/CSET/operators/filters.py

```diff
@@ -93,3 +150,78 @@ def filter_multiple_cubes(
             "The constraints don't produce a single cube per constraint."
         ) from err
     return filtered_cubes
+
+
+def generate_mask(
+    mask_field: Union[iris.cube.Cube, iris.cube.CubeList],
+    condition: str,
+    value: float,
+) -> Union[iris.cube.Cube, iris.cube.CubeList]:
+    """Generate a mask to remove data not meeting conditions.
+
+    Parameters
+    ----------
+    mask_field: iris.cube.Cube | iris.cube.CubeList
+        The field to be used for creating the mask.
+    condition: str
+        The type of condition applied, six available options:
+        '==','!=','<','<=','>', and '>='.
+    value: float
+        The value on the other side of the condition.
+
+    Returns
+    -------
+    Masks meeting the conditions applied.
+
+    Return type
+    -----------
+    iris.cube.Cube | iris.cube.CubeList
+
+    Raises
+    ------
+    ValueError: Unexpected value for condition. Expected ==, !=, >, >=, <, <=
+        Raised when condition is not supported.
+
+    Notes
+    -----
+    The mask is created in the opposite sense to numpy.ma.masked_arrays. This
+    method was chosen to allow easy combination of masks together outside of
+    this function using misc.addition or misc.multiplication depending on
+    applicability. The combinations can be of any fields such as orography >
+    500 m, and humidity == 100 %.
+
+    The conversion to a masked array occurs in the apply_mask routine, which
+    should happen after all relevant masks have been combined.
+
+    The same condition and value will be used when masking multiple cubes.
+
+    Examples
+    --------
+    >>> land_mask = generate_mask(land_sea_mask,'==',1)
+    """
+    mask_list = iris.cube.CubeList()
+    for cube in iter_maybe(mask_field):
+        masks = cube.copy()
+        masks.data = np.zeros(masks.data.shape)
+        if condition == "==":
+            masks.data[cube.data == value] = 1
+        elif condition == "!=":
+            masks.data[cube.data != value] = 1
+        elif condition == ">":
+            masks.data[cube.data > value] = 1
+        elif condition == ">=":
+            masks.data[cube.data >= value] = 1
+        elif condition == "<":
+            masks.data[cube.data < value] = 1
+        elif condition == "<=":
+            masks.data[cube.data <= value] = 1
+        else:
+            raise ValueError("""Unexpected value for condition. Expected ==, !=,
```

Comment:
Include the given condition in the error message.
```suggestion
            raise ValueError(f"Unexpected value for condition. Expected ==, !=, >, >=, <, <=. Got {condition}")
        masks.rename(f"mask_for_{cube.name()}_{condition}_{value}")
```